### PR TITLE
Flux Volume Tests

### PR DESCRIPTION
**Pull request by jleeh**

### CODE REVIEW

**Review by skudasov — COMMENTED** [review]
Nice work generalizing and refactoring that out!
Please, don't forget to adjust the log level after the metrics are fine and change Makefile target.

File: suite/performance/flux.go [hunk 1]
```diff
@@ -412,14 +441,28 @@ func (f *FluxTest) calculateLatencies(b ginkgo.Benchmarker) error {
 		for contract, contractResults := range testResults {
 			for node, nodeResults := range contractResults {
 				if nodeResults.StartTime.IsZero() {
-					log.Info().Int64("Round ID", roundID).Str("Contract", contract.Address()).Str("Node", node.URL()).Msg("Start time zero")
+					log.Info().
```
Comment (jleeh) [inline suite/performance/flux.go:444]: I put these log lines in for debugging, if keeping probably makes sense to change to Warn level.

File: suite/performance/suite_test.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 
 func TestContracts(t *testing.T) {
 	RegisterFailHandler(Fail)
-	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr}).Level(zerolog.InfoLevel)
```
Comment (jleeh) [inline suite/performance/suite_test.go:16]: Wider point, but we should be able to set log level in config.

File: contracts/ethereum_contracts.go [hunk 3]
```diff
@@ -364,17 +364,14 @@ func (f *FluxAggregatorRoundConfirmer) ReceiveBlock(block *types.Block) error {
 	if err != nil {
 		return err
 	}
-	fluxLog := log.Info().
```
Comment (jleeh) [inline contracts/ethereum_contracts.go:356]: Keep these log lines, change to debug.

File: suite/performance/flux.go [hunk 4]
```diff
@@ -192,20 +199,19 @@ func (f *FluxTest) deployContract(contractChan chan<- contracts.FluxAggregator)
...
 		for _, node := range f.chainlinkClients {
 			node := node
 			g.Go(func() error {
 				return f.createChainlinkJob(contract, node, jobsChan)
 			})
 		}
-		if err := g.Wait(); err != nil {
-			return err
-		}
+	}
```
Comment (jleeh) [inline suite/performance/flux.go:212]: Change this back. I did this specifically as too many requests at once will cause issues with file handles:
```
E0823 11:02:30.144049   25198 portforward.go:306] error accepting connection on port 45607: accept tcp4 127.0.0.1:45607: accept4: too many open files
E0823 11:02:30.144057   25198 portforward.go:306] error accepting connection on port 37949: accept tcp4 127.0.0.1:37949: accept4: too many open files
11:02AM INF Creating Job Node URL=http://127.0.0.1:37949
E0823 11:02:30.144097   25198 portforward.go:306] error accepting connection on port 37631: accept tcp4 127.0.0.1:37631: accept4: too many open files
11:02AM INF Creating Job Node URL=http://127.0.0.1:43789
```

Comment (skudasov) [reply suite/performance/flux.go:212]: Ok, nice that you have linux to check that. Maybe adding more descriptors in CI can work later.

**Pull request merged**